scribble-lib/scribble/scribble-common.js

```diff
@@ -3,67 +3,27 @@
 // Page Parameters ------------------------------------------------------------
 
 var page_query_string = location.search.substring(1);
-
-var page_args =
-  ((function(){
-      if (!page_query_string) return [];
-      var args = page_query_string.split(/[&;]/);
-      for (var i=0; i<args.length; i++) {
-        var a = args[i];
-        var p = a.indexOf('=');
-        if (p >= 0) args[i] = [a.substring(0,p), a.substring(p+1)];
-        else args[i] = [a, false];
-      }
-      return args;
-    })());
+const page_args = new URLSearchParams(location.search);
 
 function GetPageArg(key, def) {
-  for (var i=0; i<page_args.length; i++)
-    if (page_args[i][0] === key) {
-      try {
-        return decodeURIComponent(page_args[i][1]);
-      } catch (e) {
-        if (e instanceof URIError) {
-          return page_args[i][1];
-        } else {
-          throw e;
-        }
-      }
-    }
-  return def;
+  return page_args.get(key) || def;
 }
 
 function MergePageArgsIntoLink(a) {
-  if (page_args.length == 0 ||
-      (!a.attributes["data-pltdoc"]) || (a.attributes["data-pltdoc"].value == ""))
+  if (page_args.size === 0 ||
+      (!a.attributes["data-pltdoc"]) ||
+      (a.attributes["data-pltdoc"].value === ""))
```

Comment:
use `dataset` here? 

```javascript
(!a.dataset.pltdoc || a.dataset.pltdoc === "")
```
or even just
```javascript
!a.dataset.pltdoc
```
because the result of accessing dataset value will only be `undefined` or a string.

Reply:
Updated.